### DISCUSSION TITLE
Support for ATtiny406?

### DISCUSSION THREAD

**Discussion by sdalcorn** [SpenceKonde/ATTinyCore#814] [General]
OK, I have read everywhere and searched the discussions.

Is there support for ATtiny406? Do I simply use a different variation?

Thanks,

Sean

**Comment by hmeijdam** [comment]
You have been looking in the wrong town.
Look at [megaTinyCore](https://github.com/SpenceKonde/megaTinyCore)

**Reply by sdalcorn** [reply]
Thank you for your courteous reply!

**Comment by SpenceKonde** [comment]
What you *didn't* do is read the file that describes that. It's called "README", but the name doesn't seem to have worked. 

We have a list of supported devices:

https://github.com/SpenceKonde/ATTinyCore#supported-devices

Is the 406 on there? No. 

And look, if you keep scrolling down there's an "Unsupported Devices" list - and guess what, the first entry on there is for tinyAVR 0/1/2-series, which your part is, and a link to the core I created and maintain for those parts!

I put a lot of effort into the docs on the cores, and sometimes I don't think anyone reads them. (particularly mTC and DxC, because I can write once and it applies to 12-34 of the modern tinyAVRs and/or 34 Dx-series and/or 9 EA-series - 9-68 parts across 2 cores can be covered by one document for many peripherals; there are only 21 classic tinyAVR parts for this core to support).

**Reply by bxparks** [reply]
@SpenceKonde : Well... in their defense, the vast majority of README files on GitHub projects are inadequate, poorly organized, and not worth reading. I also remember a time, maybe 20-30 years ago, when corporate lawyers took over the README files, and shoved completely worthless legal crap into them. Your README is quite rare in its length and comprehensiveness. I know I have missed things in there, until I re-read it multiple times, because it has so much information.

But you are correct, in general, hardly anyone reads the README files. I get questions all the time on my projects, for things which are explained right there, on the first page of the project, right in the README that I spent weeks writing and rewriting. All they have to do is search or scroll down, but nope, they don't. It's a burden that we maintainers have to bear.

**Reply by SpenceKonde** [reply]
> the vast majority of README files on GitHub projects are inadequate, poorly organized, and not worth reading.

You forgot one - "unmodified from the repo they forked it from, even though they may have modified every other file" (Ex: every halfway popular arduino library). (this is unspeakably frustrating to me. Nobody ever says where they got the library. Like, they specify the name... not the version number - not that that would help because most repos, even "actively maintained" ones are sloppy about that. (so if it was 1.0.5 when it the ancestral repo died, there might be a fork that was maintaining it on and off for years and never thought to change the version, so version 1.0.5 could mean anything from pulling the repo that morning, to the 5-year old dead 1.0.5 release of the ancestral library. And some ambitious fellow might have declared his fork to be 2.0.0, while two other casual maintainers of forks use 1.x.x numbers if they make an attempt to number versions) - okay, so we have a name... and there are 30 github repos with that name, and they're all forks of the same thing, more than one is being actively maintained, and the ancestral one is long dead. Nobody updates their docs, nobody has a bloody changelog, and other people are finding their code and, not knowing better, using it! Some of these may be intended for public consumption, but most aren't - though *basically nobody who forks a repo for personal use bothers to change the readme. Just one line at the top saying it's modified, you aren't intending it for public consumption, you don't update docs, so others should please use the upstream repo - that'd do it. Changing the name/description is even better. But no! Nobody does that! So which of the actively maintained ones is the guy using? (A: None of them, some other one that hadn't been touched in years cause that's what the guys blog post, written uh, years ago, said to use. )

**Reply by bxparks** [reply]
This is getting slightly off topic, but a contributing factor to the stale README problem on GitHub is that GitHub makes no distinction among the different reasons that people fork projects. There are at least 3 different reasons:

1) To create a PR on the original project, GitHub forces you to fork the project, then push the commit into the fork, before sending off the pull request. Most of the time, people don't bother deleting the fork. I would say that this is probably the majority of the forks on GitHub, and for these, there is no reason to change the README. In fact, changing the README should not be done, because it would be included in the PR.

2) Sometimes, I fork a copy of a project, just to learn from it or play around with it, and I have no intention of creating a PR to the original repo, or publishing my own fork. Out of sheer laziness, I would not update the README of my personal fork of the original project.

3) The more substantial reason to create a GitHub fork is to actually fork the original project and continue maintaining the fork. Perhaps the original project is no longer maintained. Or the fork wants to go off in a different direction, feature-wise or architecturally. In this case, they should probably update the README for the fork.

GitHub provides no way to distinguish between the various cases. In fact, when you do a search on GitHub, it completely ignores all the forks, so it seems like GitHub assumes that the vast majority of forks are for reasons (1) and (2). It has very poor support for case (3), an actual fork.
